### DISCUSSION TITLE
Loading/rendering hundreds/thousands of entities.

### DISCUSSION THREAD

**Discussion by RonaldBunk** [Minestom/Minestom#2462] [Help]
I am looking into creating a gamemode for my server that could potentially need hundreds if not thousands of entities at close proximity.

Any idea if Minestom can handle that, and if so, any idea's about how?

**Comment by LooFifteen** [comment]
The only way to know is to try. I suspect network bandwidth and client rendering will become a problem before Minestom does.

**Reply by RonaldBunk** [reply]
I expected as much, but I am not sure how to test it to begin with.

Even if it would work for me, it might not for someone else; After all I do have a pretty beefy PC.

Any clue for tests, and possible optimizations if needed (probably)?

**Reply by LooFifteen** [reply]
The only way to know is to spawn thousands of entities and flamegraph it.

**Reply by ZakShearman** [reply]
Minestom will 100% handle it. I've done thousands with TowerDefence and it's been a couple % points of tick. Your client or bandwidth will definitely will be the first to die.

**Reply by RonaldBunk** [reply]
> Minestom will 100% handle it. I've done thousands with TowerDefence and it's been a couple % points of tick. Your client or bandwidth will definitely will be the first to die.

Good to know. I've tried it and it's definitely very laggy after thousands are spawned to close proximity but I feel like that optimizations might make it bearable.

Thanks for the insights.
